bin/php-language-server.php

```diff
@@ -10,6 +12,21 @@
     }
 }
 
-$server = new LanguageServer(new ProtocolStreamReader(STDIN), new ProtocolStreamWriter(STDOUT));
+if (count($argv) >= 3 && $argv[1] == '--tcp') {
+    $address = $argv[2];
+    $socket = stream_socket_client('tcp://' . $address, $errno, $errstr);
```

Comment:
I don't get the second part of your sentence, why does it have to stop listening? I though it would then solely communicate through the socket?